pkg/policy/repository.go

```diff
@@ -697,9 +707,16 @@ func (p *Repository) resolvePolicyLocked(securityIdentity *identity.Identity) (*
 	// to not have to iterate through the entire rule list multiple times and
 	// perform the matching decision again when computing policy for each
 	// protocol layer, which is quite costly in terms of performance.
+
+	lbls := securityIdentity.LabelArray
+	policyCtx := policyContext{
```

Comment:
rather than passing this down, just extract it from the labels in computePolicyEnforcementAndRules